### PR TITLE
document how to use HeliosSoloDeployment in tests

### PR DESCRIPTION
**Pull request by mattnworb**
This is just a start.

### CODE REVIEW

**Comment by mattnworb** [comment]
@negz @davidxia @gimaker @rohansingh

File: docs/integration_tests_with_helios_solo.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+Since Helios 0.8.662, [helios-solo][] can also be run directly from within an
```
Comment (mattnworb) [inline docs/integration_tests_with_helios_solo.md]: should bump this to whatever the next release is after the few in-progress improvements/PRs are merged

File: docs/integration_tests_with_helios_solo.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+client that talks to the `HeliosDeployment` and cleans it up after the test is
+finished.
+
+Then modify the `TemporaryJobs` declaration to use the client that points to
+your `HeliosSoloDeployment`:
+
+```java
+@Rule
+public TemporaryJobs temporaryJobs = TemporaryJobs.builder()
+  .client(soloResource.client())
```
Comment (davidxia) [inline docs/integration_tests_with_helios_solo.md]: Either change `soloResource` to `solo` or the `solo` on line 23 to `soloResource` to be consistent.

File: docs/integration_tests_with_helios_solo.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+#### ClassRule vs Rule
+The example above will set up `HeliosSoloDeployment` once for all of the tests
+in your `Test` class. To instead set up (and tear down) a
+`HeliosSoloDeployment` for *each individual* test in the class, change the
+`@ClassRule` to a `@Rule` and remove the `static` modifier. 
+
+Since there is a bit of set up work in creating a `HeliosSoloDeployment`, this
+will add a lot of overhead to each test. We recommend using
+`HeliosSoloDeployment` as a `@ClassRule`.
```
Comment (davidxia) [inline docs/integration_tests_with_helios_solo.md]: I'm trying to figure out how to create a test suite rule.

File: docs/integration_tests_with_helios_solo.md (hunk 2 above)
Comment (mattnworb) [inline docs/integration_tests_with_helios_solo.md]: good catch, thanks

**Comment by mattnworb** [comment]
Fixed some awkwardness in the wording and added a note about the `localhost`/`127.0.0.1` problem. Please review!

**Comment by mattnworb** [comment]
@davidxia I have a feeling that won't be possible without a lot of junit hacking.

Assuming you are looking to avoid the time-wait of starting up helios-solo for each test class, I wonder if revamping how HeliosSoloDeployment works could also work:
- HeliosSoloDeployment uses a fixed name for the container it starts (e.g. `helios-solo-auto`)
- at startup, if the `helios-solo-auto` container is running, just use it
- otherwise create and start that container
- after 10 minutes (or some other value) of non-use, the container kills itself

**Comment by davidxia** [comment]
I think the solo container should be a unique name per test run, otherwise other projects' ITs running on the same host will end up using the same solo container.

**Comment by davidxia** [comment]
:+1:

**Comment by davidxia** [comment]
@mattnworb Should the docs say to be aware of `HELIOS_HOST_FILTER`? https://github.com/spotify/helios/blob/master/helios-testing/src/test/java/com/spotify/helios/testing/HeliosSoloDeploymentTest.java#L66

**Comment by mattnworb** [comment]
@davidxia good point, they should. And/or we could hack TemporaryJobs to ignore filters when using HeliosSoloDeployment as the two concepts don't make sense together.

**Comment by davidxia** [comment]
The latter makes more sense.

**Pull request merged**